src/openforms/submissions/serializers.py

```diff
@@ -3,25 +3,53 @@
 type model fields.
 """
 
+from drf_polymorphic.serializers import PolymorphicSerializer
 from rest_framework import serializers
 
 # import from `constants` to avoid circ. import:
 from openforms.authentication.constants import AuthAttribute
 from openforms.authentication.registry import register
 
 
-class CoSignDataSerializer(serializers.Serializer):
+class PluginMixin:
+    def get_fields(self):
+        fields = super().get_fields()  # type: ignore
+        fields["plugin"].choices = register.get_choices()
+        return fields
+
+
+class CoSignV1DataSerializer(PluginMixin, serializers.Serializer):
     plugin = serializers.ChoiceField(choices=())
     identifier = serializers.CharField()
     representation = serializers.CharField(required=False, allow_blank=True, default="")
     co_sign_auth_attribute = serializers.ChoiceField(choices=AuthAttribute.choices)
     # TODO: validate fields shape depending on value of plugin (polymorphic serializer)
-    fields = serializers.DictField(
+    fields = serializers.DictField(  # type: ignore
         child=serializers.CharField(allow_blank=True),
         required=False,
         allow_null=True,
     )
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.fields["plugin"].choices = register.get_choices()
+
+class CoSignV2DataSerializer(PluginMixin, serializers.Serializer):
+    plugin = serializers.ChoiceField(choices=())
+    attribute = serializers.ChoiceField(choices=AuthAttribute.choices)
+    value = serializers.CharField()
+    cosign_date = serializers.DateTimeField()
+
+
+class CoSignDataSerializer(PolymorphicSerializer):
+    version = serializers.ChoiceField(choices=("v1", "v2"))
```

Comment:
can't use integers (yet) because the type annotation for the mapping expects strings